client/my-sites/email/titan-mail-quantity-selection/index.jsx

```diff
@@ -0,0 +1,233 @@
+import Main from 'calypso/components/main';
+import SectionHeader from 'calypso/components/section-header';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import { emailManagement } from 'calypso/my-sites/email/paths';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { recordTracksEvent as recordTracksEventAction } from 'calypso/state/analytics/actions';
+import { titanMailMonthly } from 'calypso/lib/cart-values/cart-items';
+import {
+	getDomainsBySiteId,
+	hasLoadedSiteDomains,
+	isRequestingSiteDomains,
+} from 'calypso/state/sites/domains/selectors';
+import { getDomainsWithForwards } from 'calypso/state/selectors/get-email-forwards';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import Notice from 'calypso/components/notice';
+import AddEmailAddressesCardPlaceholder from 'calypso/my-sites/email/gsuite-add-users/add-users-placeholder';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { hasTitanMailWithUs, getMaxTitanMailboxCount } from 'calypso/lib/titan';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class TitanMailQuantitySelection extends React.Component {
+	state = {
+		quantity: 1,
+	};
+
+	isMounted = false;
+
+	componentDidMount() {
+		this.isMounted = true;
+	}
+
+	componentWillUnmount() {
+		this.isMounted = false;
+	}
+
+	recordClickEvent = ( eventName ) => {
+		const { recordTracksEvent, selectedDomainName } = this.props;
+		recordTracksEvent( eventName, {
+			domain_name: selectedDomainName,
+			quantity: this.state.quantity,
+		} );
+	};
+
+	goToEmail = () => {
+		page(
+			emailManagement(
+				this.props.selectedSite.slug,
+				this.props.selectedDomain?.name,
+				this.props.currentRoute
+			)
+		);
+	};
+
+	handleCancel = () => {
+		this.recordClickEvent(
+			'calypso_email_management_titan_quantity_selection_cancel_button_click'
+		);
+		this.goToEmail();
+	};
+
+	getCartItem = () => {
+		const { selectedDomain } = this.props;
+		let quantity = this.state.quantity;
+
+		if ( hasTitanMailWithUs( selectedDomain ) ) {
+			quantity += getMaxTitanMailboxCount( selectedDomain );
+		}
+
+		return titanMailMonthly( { domain: selectedDomain.name, quantity } );
+	};
+
+	handleContinue = () => {
+		const { selectedSite } = this.props;
+
+		this.recordClickEvent(
+			'calypso_email_management_titan_quantity_selection_continue_button_click'
+		);
+
+		this.props.shoppingCartManager
+			.addProductsToCart( [
+				fillInSingleCartItemAttributes( this.getCartItem(), this.props.productsList ),
+			] )
+			.then( () => this.isMounted && page( '/checkout/' + selectedSite.slug ) );
+	};
+
+	onQuantityChange = ( e ) => {
```

Comment:
Small nitpick, but it's best for readability to avoid using one-letter variable names like this. `event` should be a better choice. Or you can even just destructure it right there: `onQuantityChange( { target } ) => {}` :)

Reply:
Hmm... I would generally agree with your comment, but in this context I feel like `e` is kinda like `i` in a for loop.